staging/src/k8s.io/apiserver/pkg/registry/rest/resttest/resttest.go

```diff
@@ -749,6 +823,38 @@ func (t *Tester) testUpdateIgnoreClusterName(obj runtime.Object, createFn Create
 // =============================================================================
 // Deletion tests.
 
+func (t *Tester) testDeleteDryRunNoGraceful(obj runtime.Object, createFn CreateFunc, getFn GetFunc, isNotFoundFn IsErrorFunc) {
```

Comment:
Why does dry run deletion need its own function?

Reply:
Because the non-dry run version verifies that the object has been removed (this one should probably verify that the object still exists)

Reply:
Would we end up with easier to maintain code if they checked opts before doing that, rather than duplicating the functions?

Reply:
(I expect yes?)

Reply:
I've changed it because I don't have a strong opinion. Sometimes it's easier to maintain mostly-copy-pasted code than factorizing diverging code with lots of if-statements. In this case it's just one if-statement so I don't really care :-)